### PR TITLE
CLOUDP-98166: Fix cloud manager e2e cluster tests

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-98166

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by colm-quinn — DISMISSED** [review]
LGTM!

File: e2e/cloud_manager/helper_test.go [hunk 1]
```diff
@@ -64,9 +69,20 @@ func automationServerHostname(cliPath string) (string, error) {
 	if servers.TotalCount == 0 {
 		return "", errors.New("no server available")
 	}
+	sort.Sort(byLastConf(*servers))
```
Comment (gssbzn) [inline e2e/cloud_manager/helper_test.go:72]: flaky, we could pick a stale server

File: e2e/cloud_manager/helper_test.go [hunk 2]
```diff
@@ -151,26 +167,26 @@ func generateRSConfig(filename, hostname, clusterName, version, fcVersion string
...
-	downloadArchive := &convert.ClusterConfig{
+	cluster := &convert.ClusterConfig{
 		RSConfig: convert.RSConfig{
 			FeatureCompatibilityVersion: fcVersion,
 			Name:                        clusterName,
 			Version:                     version,
 		},
 		Config: &convert.RSConfig{
-			Name: "configRS",
+			Name: clusterName + "_configRS",
```
Comment (gssbzn) [inline e2e/cloud_manager/helper_test.go:189]: flaky, if a previous test failed to remove the cluster we would get duplicated name error here

File: e2e/cloud_manager/dbusers_test.go [hunk 3]
```diff
@@ -40,7 +41,27 @@ func TestDBUsers(t *testing.T) {
 		t.Fatalf("unexpected error: %v", err)
 	}
 
-	const dbUsersEntity = "dbusers"
+	// make sure security is enabled, this should be a no-op for cloud manager
+	t.Run("Enable security", func(t *testing.T) {
```
Comment (gssbzn) [inline e2e/cloud_manager/dbusers_test.go:45]: things strated failed as we accidentally disabled security)

Comment (gssbzn) [reply e2e/cloud_manager/dbusers_test.go:45]: I have run this now multiple times to confirm it's an idempotent operation

File: e2e/cloud_manager/dbusers_test.go [hunk 4]
```diff
@@ -11,7 +11,8 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-// +build e2e cloudmanager,remote,replica
+//go:build e2e || (cloudmanager && remote && replica) || (opsmanager && remote && replica)
```
Comment (gssbzn) [inline e2e/cloud_manager/dbusers_test.go:14]: this is the new (1.17) way of expressing build tags, IDEA is auto-adding them on save, not going to fight it

**Comment by gssbzn** [comment]
@colm-quinn cloud manager was not happy on Friday and doing some reruns found some flakyness, this should be now RFAL

**Review by colm-quinn — APPROVED** [review]
LGTM, thanks for taking the time to isolate the flakiness in this work also

**Pull request merged**